### ISSUE TITLE
Rearrange menubar sections

### ISSUE THREAD

**Issue by randomeizer** [fcpxhacks/fcpxhacks#245]
This should happen after #136 is complete I think. At the moment, the grouping of various menu bar items in FCPX Hacks is a bit arbitrary. In particular, the terms 'SHORTCUTS', 'AUTOMATION', 'TOOLS' and 'HACKS' are all pretty undescriptive and similar.

My thought is to group the items by function, or functional area. I've already started doing this to an extent with the folders in the 'plugins' directory. For example, there are several items related to importing media, several related to the Timeline, etc. It would be better to group them together, rather than having them scattered across several menus and submenus.

A possible arrangement:

----
* Open Final Cut Pro
* Open Command Editor/Show Hacks Shortcuts
----
* *MEDIA IMPORT*
* [ ] Create Optimised Media
* [ ] Create Multicam Optimised Media
* [ ] Create Proxy Media
* [x] Leave Files In Place On Import
* [x] Ignore Inserted Camera Cards ~~(Close Media Import When Card Inserted)~~
----
* *TIMELINE*
* [ ] Enable Scrolling Timeline
* [ ] Enable Playhead Lock
* [ ] Enable Shortcuts During Full Screen Playback
* [ ] Enable Background Render (0.3 secs)
* [ ] Enable Rendering During Playback
* [ ] Enable Moving Markers
* Assign Shortcuts >
  * Effects
    * Update Effects List...
    * 1: Foobar
    * 2: [Unassigned]
    * 3: [Unassigned]
    * 4: [Unassigned]
    * 5: [Unassigned]
  * Transitions >
    * Update Transitions List...
    * 1: Foobar
    * ...
  * Generators >
    * Etc...
  * Titles >
    * Etc...
----
* *CLIPBOARD*
* Local Clipboard History >
  * [x] Enabled
  * Blah Blah (+1)
  * Clear History
* Shared Clipboard History >
  * [x] Enabled
  * Blah Blah (+1)
  * Clear History
----
* *USER INPUT* (Don't love this title)
* HUD >
  * [x] Enabled
  * [x] Show Inspector
  * [x] Show Drop Targets
  * [x] Show Buttons
  * Assign Buttons >
    * 1: [Unassigned]
    * 2: [Unassigned]
    * 3: [Unassigned]
    * 4: [Unassigned]
* Voice Commands >
  * [x] Enabled
  * [x] Announcements
  * [x] Visual Alerts
  * Open Dictation Preferences...
* Touch Bar >
  * [x] Enabled
  * [ ] Mouse Location
  * [x] Top Center of Timeline
  * TIP: ...
----
* Preferences >
  * Etc...
* Quit
----

Open to suggestions of course, but for me, something like this keeps the various related menu items together better.

**Comment by latenitefilms** [comment]
Sounds great to me!

The only reason the menubar is currently ordered the way it is, is because a lot of people were terrified of installing something that "messes" with the internals of Final Cut Pro. By separating by "Hacks" and the "Automation" features - it gave people a lot more confidence to install the software, as they didn't have to use the "Hacks" if they didn't want to.

But maybe we just have a "Enabled Advanced Features" toggle that turns on or off all the "Hacks"?

I was thinking also, maybe to make the whole thing a bit more user friendly, we introduce a "welcome screen" (similar to DaVinci Resolve) when you first run the application and this "Enabled Advanced Features" option could be part of this?

**Comment by randomeizer** [comment]
Yeah, sounds fair enough.

Maybe create the Welcome Screen as a separate issue?

On Sat, 21 Jan 2017 at 10:54 am, Chris Hocking <notifications@github.com>
wrote:

Sounds great to me!


The only reason the menubar is currently ordered the way it is, is because
a lot of people were terrified of installing something that "messes" with
the internals of Final Cut Pro. By separating by "Hacks" and the
"Automation" features - it gave people a lot more confidence to install the
software, as they didn't have to use the "Hacks" if they didn't want to.


But maybe we just have a "Enabled Advanced Features" toggle that turns on
or off all the "Hacks"?


I was thinking also, maybe to make the whole thing a bit more user
friendly, we introduce a "welcome screen" (similar to DaVinci Resolve) when
you first run the application and this "Enabled Advanced Features" option
could be part of this?




—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/fcpxhacks/fcpxhacks/issues/245#issuecomment-274218148>,
or mute the thread

**Comment by latenitefilms** [comment]
FYI: I'm also thinking maybe the current menubar should be labelled "FCPX", and we make another menubar called "CommandPost", opening up the possibility of adding additional "apps" in the future (such as Motion, iMovie and Adobe Creative Cloud app's, etc.). Maybe also rather than using text all all, we ONLY use icons?

**Comment by randomeizer** [comment]
I don't use the text label for the menu bar, ever.

However, what I'm thinking is that the contents of the menu bar changes
depending on which app you are currently using, rather than having multiple
menu bar icons.

On Wed, 25 Jan 2017 at 11:35 pm, Chris Hocking <notifications@github.com>
wrote:

FYI: I'm also thinking maybe the current menubar should be labelled "FCPX",
and we make another menubar called "CommandPost", opening up the
possibility of adding additional "apps" in the future (such as Motion,
iMovie and Adobe Creative Cloud app's, etc.). Maybe also rather than using
text all all, we ONLY use icons?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/fcpxhacks/fcpxhacks/issues/245#issuecomment-275108994>,
or mute the thread
